extensions/panache/mongodb-panache/deployment/src/main/java/io/quarkus/mongodb/panache/deployment/PanacheResourceProcessor.java

```diff
@@ -130,5 +140,39 @@ void build(CombinedIndexBuildItem index,
                 }
             }
         }
+
+        // manage @BsonProperty for the @ProjectionFor annotation
+        Map<String, String> propertyMapping = new HashMap<>();
+        for (AnnotationInstance annotationInstance : index.getIndex().getAnnotations(DOTNAME_PROJECTION_FOR)) {
+            Type targetClass = annotationInstance.value().asClass();
+            ClassInfo target = index.getIndex().getClassByName(targetClass.name());
+            for (FieldInfo fieldInfo : target.fields()) {
+                if (fieldInfo.hasAnnotation(DOTNAME_BSON_PROPERTY)) {
+                    AnnotationInstance bsonProperty = fieldInfo.annotation(DOTNAME_BSON_PROPERTY);
+                    propertyMapping.put(fieldInfo.name(), bsonProperty.value().asString());
+                }
+            }
+            for (MethodInfo methodInfo : target.methods()) {
+                if (methodInfo.hasAnnotation(DOTNAME_BSON_PROPERTY)) {
+                    AnnotationInstance bsonProperty = methodInfo.annotation(DOTNAME_BSON_PROPERTY);
+                    propertyMapping.put(methodInfo.name(), bsonProperty.value().asString());
+                }
+            }
+        }
+        for (AnnotationInstance annotationInstance : index.getIndex().getAnnotations(DOTNAME_PROJECTION_FOR)) {
+            ClassInfo info = annotationInstance.target().asClass();
+            for (FieldInfo fieldInfo : info.fields()) {
```

Comment:
Does this support inheritance? Should it? If not, it should be documented.

Reply:
I think inheritance will works as long as the user also annotate the parent class with `@ProjectionFor`. This should be tested ...

Reply:
I added a line inside the guide about inheritance